### ISSUE TITLE
Proposal: use generic "inflate" function in Ansible modules

### ISSUE THREAD

**Issue by zikalino** [Azure/magic-module-specs#39]
Since the implementation of parameter transformation may be missing for more complex cases including arrays of complex structures, maybe we could consider using generic "inflate" function which can perform appropriate transformation based on arg_spec.
Perhaps that would simplify a lot Ansible module generation.

More discussion here:
https://github.com/Azure/magic-module-specs/issues/22

**Comment by JunyiYi** [comment]
I agree with you because MM has not done many cases for those transformations. We can do it in a shared function and actually I prefer to do that for three reasons:

1. it makes module code looks simpler
2. it improves the code quality because I am not an expert of Ansible, I might miss some details in MM;
3. easy to maintain in the future, it reduces the chance to change MM

The reason why Terraform does not use this method is because Hashicorp against this kind of coding style, they think it might break some resources if you change the shared part.
